service/src/main/java/bio/terra/landingzone/service/landingzone/azure/LandingZoneService.java

```diff
@@ -111,6 +116,56 @@ public void deleteLandingZone(String landingZoneId) {
     throw new LandingZoneDeleteNotImplemented("Delete operation is not implemented");
   }
 
+  public LandingZoneResourcesByPurpose listResourcesWithPurposes(
+      AzureCloudContext azureCloudContext) {
+    LandingZoneManager landingZoneManager =
+        landingZoneManagerProvider.createLandingZoneManager(azureCloudContext);
+
+    return LandingZoneResourcesByPurpose.builder()
+        .generalResources(listGeneralResourcesWithPurposes(landingZoneManager))
+        .subnetResources(listVNetResourcesWithPurposes(landingZoneManager))
+        .build();
+  }
+
+  private Map<String, List<LandingZoneResource>> listGeneralResourcesWithPurposes(
+      LandingZoneManager landingZoneManager) {
+    var deployedResources = landingZoneManager.reader().listResourcesWithPurpose();
+
+    return deployedResources.stream()
+        .map(
+            dp ->
+                LandingZoneResource.builder()
+                    .resourceId(dp.resourceId())
+                    .resourceType(dp.resourceType())
+                    .tags(dp.tags())
+                    .region(dp.region())
+                    .build())
+        .collect(
+            Collectors.groupingBy(
+                r -> r.tags().get(LandingZoneTagKeys.LANDING_ZONE_PURPOSE.toString())));
+  }
+
+  private Map<String, List<LandingZoneSubnetResource>> listVNetResourcesWithPurposes(
+      LandingZoneManager landingZoneManager) {
+    var deployedVNetResources = landingZoneManager.reader().listVNetResourcesWithSubnetPurpose();
+    List<LandingZoneSubnetResource> deployedSubnetList = new ArrayList<>();
+    deployedVNetResources.stream()
```

Comment:
Minor - don't have to use, but was wondering if this could be a `flatMap` :) 

```
    return deployedVNetResources.stream()
            .flatMap(dp ->
                    dp.subnetIdPurposeMap().entrySet().stream()
                            .map(e ->
                                    LandingZoneSubnetResource.builder()
                                      .name(e.getValue().name())
                                      .subnetPurpose(e.getKey().toString())
                                      .vNetId(dp.Id())
                                      .region(dp.region())
                                      .build()))
            .collect(Collectors.groupingBy(LandingZoneSubnetResource::getSubnetPurpose));
```